src/Titon/Console/Command/AbstractCommand.hh

```diff
@@ -0,0 +1,344 @@
+ */
+abstract class AbstractCommand implements Command {
+
+    /**
+     * Bag container holding all registered `Argument` objects
+     *
+     * @var \Titon\Console\InputBag<Argument>
+     */
+    protected InputBag<Argument> $arguments;
+
+    /**
+     * The description of the command used when rendering its help screen.
+     *
+     * @var string
+     */
+    protected string $description = '';
+
+    /**
+     * Bag container holding all registered `Flag` objects
+     *
+     * @var \Titon\Console\InputBag<Flag>
+     */
+    protected InputBag<Flag> $flags;
+
+    /**
+     * The `Input` object containing all registered and parsed command line
+     * parameters.
+     *
+     * @var \Titon\Console\Input
+     */
+    protected Input $input;
+
+    /**
+     * The name of the command passed into the command line.
+     *
+     * @var string
+     */
+    protected string $name = '';
+
+    /**
+     * Bag container holding all registered `Option` objects
+     *
+     * @var \Titon\Console\InputBag<Option>
+     */
+    protected InputBag<Option> $options;
+
+    /**
+     * The `Output` object to handle output to the user.
+     *
+     * @var \Titon\Console\Output
+     */
+    protected Output $output;
+
+    /**
+     * Construct a new instance of a command.
+     */
+    public function __construct(Input $input, Output $output) {
+        $this->input = $input;
+        $this->output = $output;
+        $this->arguments = new InputBag();
+        $this->flags = new InputBag();
+        $this->options = new InputBag();
+    }
+
+    /**
+     * Add a new `Argument` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Argument $argument
+     *
+     * @return $this
+     */
+    public function addArgument(Argument $argument): this {
+        $this->arguments->set($argument->getName(), $argument);
+
+        return $this;
+    }
+
+    /**
+     * Add a new `Flag` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Flag $flag
+     *
+     * @return $this
+     */
+    public function addFlag(Flag $flag): this {
+        $this->flags->set($flag->getName(), $flag);
+
+        return $this;
+    }
+
+    /**
+     * Add a new `Option` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Option $option
+     *
+     * @return $this
+     */
+    public function addOption(Option $option): this {
+        $this->options->set($option->getName(), $option);
+
+        return $this;
+    }
+
+    /**
+     * Construct and return a new `Menu` object given the choices and display
+     * message.
+     *
+     * @param Map<mixed, mixed> $choices    Accepted values
+     * @param string $message               The message to display before the choices
+     *
+     * @return \Titon\Console\UserInput\Menu
+     */
+    protected function confirm(Map<mixed, mixed> $choices, string $message = ''): Menu {
```

Comment:
We could use generics, since confirm will always be the same, but Menu for example accepts both integer and string keys. Whichever is the key value, when the user selects an option, the key is returned. So the developer could either receive input of the integer key or the string they designated as the choice.